sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRepositoryLiveTests.cs

```diff
@@ -0,0 +1,277 @@
+        {
+            // Arrange
+            var client = CreateClient();
+            var repository = client.GetRepository(_repositoryName);
+            int pageSize = 1;
+            int minExpectedPages = 2;
+
+            AsyncPageable<ArtifactManifestProperties> manifests = repository.GetManifestsAsync();
+            string firstDigest = null;
+            string secondDigest = null;
+            int artifactCount = 0;
+            await foreach (var manifest in manifests)
+            {
+                if (artifactCount == 0)
+                {
+                    firstDigest = manifest.Digest;
+                }
+
+                if (artifactCount == 1)
+                {
+                    secondDigest = manifest.Digest;
+                    break;
+                }
+
+                artifactCount++;
+            }
+
+            // Act
+            manifests = repository.GetManifestsAsync();
+            var pages = manifests.AsPages($"</acr/v1/{_repositoryName}/_manifests?last={firstDigest}&n={pageSize}>");
+
+            int pageCount = 0;
+            Page<ArtifactManifestProperties> firstPage = null;
+            await foreach (var page in pages)
+            {
+                if (pageCount == 0)
+                {
+                    firstPage = page;
+                }
+
+                Assert.LessOrEqual(page.Values.Count, pageSize);
+                pageCount++;
+            }
+
+            // Assert
+            Assert.AreNotEqual(null, firstPage);
+            Assert.AreEqual(secondDigest, firstPage.Values[0].Digest);
+            Assert.GreaterOrEqual(pageCount, minExpectedPages);
+        }
+
+        [RecordedTest]
+        public async Task CanGetManifestsOrdered()
+        {
+            // Arrange
+            string repositoryName = $"library/node";
+            string tag = "newest";
+            var client = CreateClient();
+            var repository = client.GetRepository(repositoryName);
+            var artifact = client.GetArtifact(repositoryName, tag);
+
+            try
+            {
+                if (Mode != RecordedTestMode.Playback)
+                {
+                    await ImportImage(repositoryName, tag);
+                }
+
+                // Act
+                AsyncPageable<ArtifactManifestProperties> manifests = repository.GetManifestsAsync(ManifestOrderBy.LastUpdatedOnDescending);
+
+                // Assert
+                string digest = null;
+                await foreach (ArtifactManifestProperties manifest in manifests)
+                {
+                    // Make sure we're looking at a manifest list, which has the tag
+                    if (manifest.References != null && manifest.References.Count > 0)
+                    {
+                        digest = manifest.Digest;
+                        Assert.That(manifest.RepositoryName.Contains(repositoryName));
+                        Assert.That(manifest.Tags.Contains(tag));
```

Comment:
suggestion: `Assert.Contains(...` provides a better assertion message on failure.

Reply:
From Intellisense, it looks like Assert.Contains() looks for an element in a collection - this is looking for a substring.  Am I missing another method on Assert?

Reply:
np! :)